client/components/language-picker/modal.jsx

```diff
@@ -255,7 +254,21 @@ export class LanguagePickerModal extends PureComponent {
 		);
 	}
 
+	/**
+	 * Only record a single search event per modal instance
+	 * otherwise we would get a search event per keystroke.
+	 */
+	recordSearchTracksEvent = () => {
+		if ( this.state.recordedTracksSearchEvent ) {
+			return;
+		}
+
+		this.setState( { recordedTracksSearchEvent: true } );
+		this.props.recordTracksEvent( 'calypso_language_picker_searched' );
```

Comment:
As far as I understand, the goal of this PR is to find out whether the user searches or not before selecting a language. 

We're achieving that by recording the `calypso_language_picker_new_language_picked` event, but I don't see the value of having a separate event that the user has used the search form. 

Could you please elaborate on that decision? Maybe I'm missing something here. Or could we just remove that event altogether?

Reply:
Maybe I am misunderstanding what @yuliyan meant in this comment: https://github.com/Automattic/wp-calypso/pull/46110#pullrequestreview-501905022

My understanding is that we want two events:

1. An event to track whenever search is used _at all_ regardless of whether the person leaves the modal with a new language selection
2. (Extra credit) An event to track whether a new language pick resulted from the user using search.

We could combine these two into a more generic event `calypso_language_picker_used` (?) with two props `searched` and `newLanguageSelected` for example that fires on the modal's close. Two separate events seems to be clearer to me though. What do you think?

Reply:
I'm sorry for the confusion. What I meant is that basically we should get [this condition removed](https://github.com/Automattic/wp-calypso/pull/46110/files#diff-aee98c4a97ee91a31bf5a3d54d45ca97R436-R442). Then it would give us the total count of the language picker interactions, and `searched` prop could be used to find what fraction of the total language picker interaction involved using the search bar.

As for the second event, I thought for some reason that it only accepts single argument. But since that's not the case, it makes sense to remove it and just pass `isClosingWithoutSelection` to the first event.